lib/charms/layer/apache_bigtop_base.py

```diff
@@ -39,12 +40,32 @@ def site_yaml(self):
     def __init__(self):
         self.bigtop_dir = '/home/ubuntu/bigtop.release'
         self.options = layer.options('apache-bigtop-base')
-        self.bigtop_version = self.options.get('bigtop_version')
-        self.bigtop_apt = self.options.get(
-            'bigtop_repo-{}'.format(utils.cpu_arch()))
-        self._bigtop_base = Path(self.bigtop_dir) / self.bigtop_version
-        self._site_yaml = self.bigtop_base / self.options.get(
-            'bigtop_hiera_siteyaml')
+        self.bigtop_version = self.options['bigtop_version']
+        self._bigtop_base = Path(self.bigtop_dir) / 'bigtop-{}'.format(
+            self.bigtop_version)
+        self._site_yaml = Path(self.bigtop_base) / self.options[
+            'bigtop_hiera_siteyaml']
```

Comment:
If we're going to fail hard on key errors for options, I'd prefer to see us raise a custom exceptions. Something like BigtopConfigError. That way, the person troubleshooting isn't stuck tracking down cryptic KeyErrors in the code. 

Also, this will melt the tests -- you'll need to replace the mock object with a dict with mock values in the setUp for each test class.